include/vigra/recursive_multi_convolution.hxx

```diff
@@ -0,0 +1,404 @@
+    {deriche_1stderiv_2nd, deriche_1stderiv_3rd, deriche_1stderiv_4th},
+    // second derivative
+    {deriche_smoothing_4th, deriche_smoothing_4th, deriche_smoothing_4th}};
+
+} // namespace detail
+
+#define VIGRA_RECURSIVE_CONVOLUTION_OPTIONS(                                   \
+    class_name, function_name, default_value, member_name, getter_setter_name) \
+    template <class Param> class_name &function_name(const Param &val) {       \
+        member_name = ParamVec(val, #class_name #function_name);               \
+        return *this;                                                          \
+    }                                                                          \
+    class_name &function_name() {                                              \
+        member_name = ParamVec(default_value, #class_name #function_name);     \
+        return *this;                                                          \
+    }                                                                          \
+    class_name &function_name(double v0, double v1) {                          \
+        member_name = ParamVec(v0, v1, #class_name #function_name);            \
+        return *this;                                                          \
+    }                                                                          \
+    class_name &function_name(double v0, double v1, double v2) {               \
+        member_name = ParamVec(v0, v1, v2, #class_name #function_name);        \
+        return *this;                                                          \
+    }                                                                          \
+    class_name &function_name(double v0, double v1, double v2, double v3) {    \
+        member_name = ParamVec(v0, v1, v2, v3, #class_name #function_name);    \
+        return *this;                                                          \
+    }                                                                          \
+    class_name &function_name(double v0, double v1, double v2, double v3,      \
+                              double v4) {                                     \
+        member_name =                                                          \
+            ParamVec(v0, v1, v2, v3, v4, #class_name #function_name);          \
+        return *this;                                                          \
+    }                                                                          \
+    typename ParamVec::p_vector get##getter_setter_name() const {              \
+        return member_name.vec;                                                \
+    }                                                                          \
+    void set##getter_setter_name(typename ParamVec::p_vector vec) {            \
+        member_name.vec = vec;                                                 \
+    }
+
+template <unsigned dim> class RecursiveConvolutionOptions {
```

Comment:
I think you can just add the function `recursiveOrder()` to the existing `ConvolutionOptions` class